src/herbivoresAndCarnivores.js

```diff
@@ -1,19 +1,46 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
```

Comment:
The `die` method currently uses `indexOf` and `splice` to remove an animal from the `Animal.alive` array. Consider using `filter` to create a new array without the current animal, which can improve performance by avoiding nested loops.